playbook.yaml

```diff
@@ -55,14 +55,42 @@
         - objects/namespace-install.yaml
         - objects/argocd_server_route.yaml
 
-    - name: Patch ArgoCD admin psw
+    - name: Get Dex Service Account Token
+      shell: oc serviceaccounts get-token argocd-dex-server
+      register: dex_sa_token
+
+    - name: Get OpenShift API server
+      shell: oc whoami --show-server
+      register: api_server
+
+    - name: Get the ArgoCD Dex Service Account
+      k8s_facts:
+        kubeconfig: "{{ kubeconfig }}"
+        kind: Route
+        name: argocd-server
+        namespace: "{{ namespace }}"
+        verify_ssl: no
+      register: server_route
+
+    - name: Set Dex Connecter fields
+      set_fact:
+        argocd_url: "https://{{ server_route.resources[0].spec.host }}"
+        issuer: "{{ api_server.stdout }}"
+        clientID: "system:serviceaccount:{{ namespace }}:argocd-dex-server"
+        clientSecret: "{{ dex_sa_token.stdout }}"
+        redirectURI: "https://{{ server_route.resources[0].spec.host }}/api/dex/callback"
+
+    - debug:
```

Comment:
Is this necessary every time we run the playbook?

Reply:
nope that shouldn't be there